### DISCUSSION TITLE
Child component wire:click complains about missing method on parent

### DISCUSSION THREAD

**Discussion by johnnuman** [livewire/livewire#2716] [1. Help] [answered]
Hi there,

I  have exhausted my search for a solution on every possible forum for past 24 hours. And now I am posting a new request for help.

Here is my setup:
I am creating a multi-step wizard style form using nested Livewire components.

- The parent

`
class Parent extends Component {
    public $currentStep = 4;

    public $wizardSteps;

    protected $listeners = [
        'stepForward' => 'nextStep',
        'stepBack' => 'prevStep',
    ];

    public function mount()
    {
        $this->wizardSteps = [
            'Terms & conditions',
            'Choose a loan type',
            'Loan requirements',
            'Assets & liabilities',
            'Income & expenditure',
            'Credit rating',
            'Payment',
            'Done'
        ];
    }

    public function render()
    {
        return view('livewire.borrower.loans.new-loan');
    }

    public function nextStep()
    {
        $this->currentStep++;
    }

    public function prevStep()
    {
        $this->currentStep--;
    }
}`

- the parent blade
`<div class="intro-y box flex-initial h-full flex-grow flex flex-col py-3 sm:pt-6 mt-5 min-h-0">
    <x-general.wizard :currentStep="$currentStep" :steps="$wizardSteps">
        <div>
            @livewire('borrower.loans.new-loan.step' . $currentStep, [], key('step-' . $currentStep))
        </div>
    </x-general.wizard>
</div>`

- the child
`class Step4 extends Component
{

    public $currentLoan;

    public $applicantsCount = 2;

    public function mount()
    {
    }

    public function render()
    {
        return view('livewire.borrower.loans.new-loan.step4');
    }

   // this does not fire.
    public function applicantsCountChanged() {
        dd('here');
    }

  // removed irrelevant code
}
`

- child blade file
` <style xmlns:x-aire="http://www.w3.org/1999/html">
    .main{
        height: inherit!important;
    }
</style>

<div class="flex flex-col flex-grow min-h-0">

    <div class="flex items-center justify-end px-4 pt-3 text-right sm:px-6 border-t">
        <x-general.button wire:click="prevStep" class="btn-secondary">{{ __('Go back') }}</x-general.button>
        <x-general.button wire:click="applicantsCountChanged" class="btn-primary text-white ml-2 " >Save & continue</x-general.button>
    </div>
</div>
 `

### Problem statement
Clicking on the button in child blade file should fire **applicantsCountChanged** method on the child component file. But it complains about the method missing on the parent component file. See screenshot
![livewire-issue](https://user-images.githubusercontent.com/33455481/114514809-1eee0280-9c55-11eb-8b36-334d74f64a85.png)

Please advise. Thank you.

**Comment by joshhanley** [comment, marked as answer]
@johnnuman you should only have one root html element in your components blade file. But in you child component you have two, `<style>` and `<div>`.

https://laravel-livewire.com/docs/2.x/rendering-components#returning-blade:~:text=Make%20sure%20your%20Blade%20view%20only%20has%20ONE%20root%20element.

What is happening, if you inspect in your browser tools, is the child component is initialising on the `<style>` element, hence why anything in the div is acting as if it is in the parent.

At the minimum, I would at least put the `<div>` first and the `<style>` second.

Hope this helps!

**Reply by calebporzio** [reply]
@joshhanley , thanks so much for helping out with this.

We've talked about throwing some kind of error when livewire detects two root elements.

Maybe we need to revisit that discussion to avoid problems like this?

The solution could be simple if we keep it in javascript land. (Easier to detect multiple nodes in JS than PHP).

Thoughts?

**Reply by joshhanley** [reply]
@calebporzio no worries! Yep agreed, it's one of the more common issues I see people run into. If there was a console error at least, then it's simpler to support. Just get them to check console and go from there 🙂

A long term solution would actually be to support multiple components, I don't know what that'd look like yet. Whether Livewire attaches itself to all root elements which share a common JS component or if you'd have to use some form of wrapper (which could cause layout issues).

That's a deeper discussion, so a console error would be the best short term I think!

**Reply by calebporzio** [reply]
Agreed!

**Comment by johnnuman** [comment]
@joshhanley Thank you so much for the spot on suggestion. Indeed it was the multiple children at the root that caused the problem. Wrapping those in `<div>` as suggested resolved the problem. Thank you.
